src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState, forwardRef } from "react";
+import cx from "classnames";
+import moment from "moment";
+import "react-dates/lib/css/_datepicker.css";
+import { DayPickerSingleDateController, DayPickerRangeController, DayOfWeekShape } from "react-dates";
+import DatePickerHeaderComponent from "./DatePickerHeader/DatePickerHeader";
+import DateNavigationItem from "./DateNavigationItem/DateNavigationItem";
+import YearPicker from "./YearPicker/YearPicker";
+import { DAY_SIZE, WEEK_FIRST_DAY } from "./constants";
+import { Moment, FocusInput, Direction, RangeDate } from "./types";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import VibeComponent from "../../types/VibeComponent";
+import styles from "./DatePicker.module.scss";
+import "react-dates/initialize";
+interface DatePickerProps extends VibeComponentProps {
+  firstDayOfWeek: DayOfWeekShape;
+  date?: Moment;
+  endDate?: Moment;
+  onPickDate: (date: Moment | RangeDate) => void;
+  hideNavigationKeys: boolean;
+  enableOutsideDays: boolean;
+  showWeekNumber: boolean;
+  daySize: number;
+  shouldBlockDay?: (date: Moment) => boolean;
+  range: boolean;
+  numberOfMonths: number;
+  shouldBlockYear?: (year: number) => boolean;
+  isOutsideRange?: (date: Moment) => boolean;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const NOOP = () => {};
+
+const DatePicker: VibeComponent<DatePickerProps, HTMLElement> = forwardRef<HTMLDivElement, DatePickerProps>(
+  (
+    {
+      id,
+      className,
+      firstDayOfWeek = WEEK_FIRST_DAY,
+      daySize = DAY_SIZE,
+      range = false,
+      shouldBlockDay,
+      shouldBlockYear,
+      numberOfMonths = 1,
+      hideNavigationKeys = false,
+      date,
+      endDate,
+      onPickDate,
+      enableOutsideDays = false,
+      showWeekNumber = false,
+      isOutsideRange,
+      "data-testid": dataTestId
+    },
+    ref
+  ) => {
+    const [focusedInput, setFocusedInput] = useState(FocusInput.startDate);
+    const [isMonthYearSelection, setIsMonthYearSelection] = useState(false);
+    const [overrideDateForView, setOverrideDateForView] = useState<Moment | null>(null);
+
+    const renderMonth = ({ month }: { month: Moment }) => {
+      return (
+        <DatePickerHeaderComponent
+          data-testid={dataTestId}
+          currentDate={month || moment()}
+          isMonthYearSelection={isMonthYearSelection}
+          onToggleMonthYearPicker={() => setIsMonthYearSelection(val => !val)}
+          hideNavigationKeys={hideNavigationKeys}
+        />
+      );
+    };
+
+    const renderDay = (day: Moment) => {
```

Comment:
We do not pass anonymous functions which implemented in our component's render function to third-party component as props.
The reason we do not do this is because this way we actually will create a new instance of renderDay every time the date picker will render and we can't fully understand how it will affect the third party component to receive different instance each time.
Please use here useCallback.